### PR TITLE
Updated PHP SDK

### PR DESCRIPTION
**Pull request by Millasdur**
Added optional parameters : memory and log_level, to the dialog endpoint
Added a proxy option
Removed audio endpoint

### CODE REVIEW

File: src/apis/Request/Request.php [hunk 1]
```diff
@@ -50,37 +51,11 @@ public function analyseText($text, $options = []) {
...
-
-    return new \RecastAI\apis\Resources\Response($responseBody);
-  }
-
-  public function converseText($text, $options = []) {
+  public function converseText($text, $options = [], $memory = [], $log_level = 0) {
     $token = array_key_exists('token', $options) ? $options['token'] : $this->token;
     $language = array_key_exists('language', $options) ? $options['language'] : $this->language;
     $conversation_token = array_key_exists('conversation_token', $options) ? $options['conversation_token'] : NULL;
-    $memory = array_key_exists('memory', $options) ? $options['memory'] : NULL;
```
Comment (mandre42) [inline src/apis/Request/Request.php]: Why is this removed ?
($memory is used in line 69)

File: src/apis/Request/Request.php [hunk 2]
```diff
@@ -91,7 +66,8 @@ public function converseText($text, $options = []) {
       "text" => $text,
       "language" => $language,
       "conversation_token" => $conversation_token,
-      "memory" => $memory
+      "memory" => $memory,
```
Comment (dbousque) [inline src/apis/Request/Request.php]: Could you set the `memory` field only if it was given as argument please ? It may replace the existing memory with an empty one otherwise

Comment (mandre42) [reply src/apis/Request/Request.php]: @dbousque No because memory is set to NULL by default

Comment (dbousque) [reply src/apis/Request/Request.php]: It is not correct to pass a `null` memory and expect nothing to happen :) According to the bot-builder API, it could very well reset the memory, it is not semantically wrong, so let's just make things easier. Someone might think it was an error to set it to `null` instead of `{}` for example, and correcting that, he would break everything.

Comment (mandre42) [reply src/apis/Request/Request.php]: Usage of these parameter is describe in the documentation (Wiki)
Regarding the BotBuilderAPI project code, the memory parameter is taken in account only if its value is "true" (so, not equal to null). This code works

**Comment by mandre42** [comment]
good for me
